qip-0015.md

```diff
@@ -0,0 +1,75 @@
+  Type: Standards Track
+  Created: 2024-09-06
+  License: BSD-2-Clause
+  Replaces: QIP-0012
+```
+
+## Abstract
+
+The UTXO set is the set of all unspent transaction outputs in the Qi ledger. Qi is a money system with cash-like properties, such as denominations that can be broken down into smaller denominations. Combining smaller denominations into larger ones can also be done with a different mechanism, but that is to be discussed in a separate QIP. This QIP proposes a mechanism to regularly remove dust UTXOs from the UTXO set.
+
+## Motivation
+
+The Qi UTXO set must be held on the disk of every node in the network in order for nodes to come into consensus, and the set can grow infinitely large. There are currently about [187.3 million UTXOs](https://www.blockchain.com/explorer/charts/utxo-count) in Bitcoin, and at ~70ish bytes per UTXO, takes up about 13GB of space on disk. 
+To build a monetary system for every individual the UTXO set could grow to as large as 20 billion for a given Zone, which is about 200 UTXOs per user for 100 million users per Zone, for example. A Zone that processes 1,000 transactions per second would allow 100  million users to send one transaction per day, and if the demand grows larger, a new Zone would be added. Qi UTXOs are about 57 bytes: 2 bytes for the leveldb prefix, 32 bytes for the txhash, 2 bytes for the index, 20 bytes for the address that may spend the UTXO, 1 byte for the denomination, and 1 byte or more for the lock (if applicable). This means that the UTXO set on disk could grow to as large as 1.14 TB per Zone. 
+Because the UTXO set can grow to be infinitely large, a well-funded attacker could launch a DDoS attack to grow the UTXO set to a size that is untenable for the average node to store on disk. In order to prevent this attack, this QIP proposes removing old, unused small-value UTXOs (below 1 Qi), such that the DDoS bloating attack would be expontentially more expensive to carry out - a bloating attack to increase the size of the UTXO set by 10 billion UTXOs would cost 10 billion Qi vs 10 million Qi, for example.  
+
+## Specification
+
+### UTXO Trimming
+
+The following table displays small-value denominations and the timeframe after which each respective UTXO should be trimmed (removed from the set). Note that denomination 0 is equivalent to 0.001 Qi or 1 "Qit" (the smallest unit of Qi).
+
+| Denomination      | Value                 | Timeframe                         
+|-------------------|-----------------------|-------------------
+| 0                 | 0.001 Qi (1 "Qit")    | 2 weeks                      
+| 1                 | 0.005 Qi (5 "Qits")   | 1 month                   
+| 2                 | 0.01 Qi (10 "Qits")   | 2 months 
+| 3                 | 0.05 Qi (50 "Qits")   | 4 months          
+| 4                 | 0.1 Qi (100 "Qits")   | 8 months         
+| 5                 | 0.25 Qi               | 10 months         
+| 6                 | 0.5 Qi                | 12 months 
```

Comment:
please drop this table from the qip (or at least the rates). Its unrelated to this qip, and could become confusing if those rates are changed elsewhere, but this qip isn't updated.

This qip should only describe the lock/unlock mechanism and its motivation.